src/num/flpmath.c

```diff
@@ -2992,3 +2992,53 @@ void md_zfftmod(unsigned int D, const long dims[D], complex float* optr, const c
 
 	md_zfftmod2(D, dims, strs, optr, strs, iptr, inv, phase);
 }
+
+extern void md_calc_fmac3_dims(long N, long squash, long dims1[N], long dims2[N], long out_dims[N]) 
+{
+	long dims[N];
+	for (int i = 0; i < N; i++) {
+		assert((dims1[i] == dims2[i]) || (1 == dims1[i]) || (1 == dims2[i]));
+		dims[i] = (1 == dims1[i]) ? dims2[i] : dims1[i];
+	}   
+	md_select_dims(N, ~squash, out_dims, dims);
+}
+
+extern void md_fmac3(long N, long dims1[N], complex float* input1, long dims2[N], complex float* input2, long out_dims[N], complex float* out_data) 
```

Comment:
Good points. I would prefer the names md_calc_zfmac_dims and md_zfmac_dims. I don't see the use for a md_calc_fmac_strs. As I understand it, md_zfmac_dims (here named md_fmac3) is just a convienince function for not having to define strides and callling md_calc_strides three times before calling md_zfmac2.